pkg/server/commands/list_objects.go

```diff
@@ -301,13 +292,15 @@ func (q *ListObjectsQuery) performChecks(ctx context.Context, req listObjectsReq
 
 func (q *ListObjectsQuery) internalCheck(
 	ctx context.Context,
+	combinedDatastore storage.RelationshipTupleReader,
 	obj *openfgapb.Object,
 	req listObjectsRequest,
 	objectsFound *uint32,
 	resultsChan chan<- string,
 ) error {
+	checkResolver := graph.NewLocalChecker(combinedDatastore, maximumConcurrentChecks)
```

Comment:
This should only be constructed once, not per `internalCheck` invocation. Constructing it each invocation is wasteful.